src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -87,20 +86,41 @@ public OpticalDuplicateFinder(final String readNameRegex, final int opticalDupli
 
         final PhysicalLocation actualKeeper = keeperOrNull(list, keeper);
 
+        final Log log = Log.getInstance(OpticalDuplicateFinder.class);
+        ProgressLogger progressLogger = new ProgressLogger(log, 10000, "examined", "ReadEnds to keeper");
+        if(length > 1000) {
+            log.info("Large duplicate set. size = " + length);
+            log.debug("About to compare to keeper:" + actualKeeper);
+        }
+
         // First go through and compare all the reads to the keeper
         if (actualKeeper != null) {
-            for (int i=0; i<length; ++i) {
+            for (int i = 0; i < length; ++i) {
                 final PhysicalLocation other = list.get(i);
                 opticalDuplicateFlags[i] = closeEnough(actualKeeper, other, distance);
+
+                // The main point of adding this log and if statement (also below) is a workaround a bug in the JVM
+                // which causes a deep exception (https://github.com/broadinstitute/picard/issues/472)
+                // it seems that this is related to https://bugs.openjdk.java.net/browse/JDK-8033717 which
+                // was closed due to non-reproducibility. We came across a bam file that evoked this error
+                // every time we tried to mark duplicates it. It had a duplicate-set of size 500,000, and
```

Comment:
`time we tried to mark duplicates it. It had a duplicate-set` -> `time we tried to duplicate mark it, it had a duplicate-set`.